File: src/YTStdTenantPlatform/entity/TenantPlatform/Notification.cs
using System;
using YTStdEntity.Attributes;

namespace YTStdTenantPlatform.Entity.TenantPlatform;

/// <summary>通知</summary>
[Entity(TableName = "notifications")]
[Index("idx_notifications_tenant_status", "tenant_ref_id", "send_status")]
public class Notification
{
    /// <summary>主键</summary>
    [Column(IsPrimaryKey = true)]
    public long Id { get; set; }

    /// <summary>关联租户 ID</summary>
    public long? TenantRefId { get; set; }

    /// <summary>模板 ID</summary>
    public long? TemplateId { get; set; }

    /// <summary>通知渠道</summary>
    [Column(Length = 32, IsRequired = true)]
    public string Channel { get; set; } = "";

    /// <summary>接收人</summary>
    [Column(Length = 255, IsRequired = true)]
    public string Recipient { get; set; } = "";

    /// <summary>主题</summary>
    [Column(Length = 255)]
    public string? Subject { get; set; }

    /// <summary>正文</summary>
    [Column(IsRequired = true)]
    public string Body { get; set; } = "";

    /// <summary>发送状态</summary>
    [Column(Length = 32, IsRequired = true)]
    public string SendStatus { get; set; } = "";

    /// <summary>发送时间</summary>
    public DateTime? SentAt { get; set; }

    /// <summary>阅读时间</summary>
    public DateTime? ReadAt { get; set; }

    /// <summary>创建时间</summary>
    public DateTime CreatedAt { get; set; }
}

File: web/tenant-platform-web/src/types/notification.ts
/** 通知模板响应 */
export interface NotificationTemplateRepDTO {
  Id: number
  TemplateCode: string
  TemplateName: string
  Channel: string
  SubjectTemplate: string | null
  BodyTemplate: string
  Status: string
  CreatedAt: string
}

/** 创建通知模板请求 */
export interface CreateNotificationTemplateReqDTO {
  TemplateCode: string
  TemplateName: string
  Channel?: string
  SubjectTemplate?: string
  BodyTemplate: string
}

/** 更新通知模板请求 */
export interface UpdateNotificationTemplateReqDTO {
  TemplateName?: string
  SubjectTemplate?: string
  BodyTemplate?: string
}

/** 通知响应 */
export interface NotificationRepDTO {
  Id: number
  TenantRefId: number | null
  TemplateId: number | null
  Channel: string
  Recipient: string
  Subject: string | null
  Body: string
  SendStatus: string
  SentAt: string | null
  ReadAt: string | null
  CreatedAt: string
}

/** 创建通知请求 */
export interface CreateNotificationReqDTO {
  TenantRefId?: number | null
  TemplateId?: number | null
  Channel?: string
  Recipient: string
  Subject?: string
  Body: string
}
